.github/workflows/license-check.yml

```diff
@@ -0,0 +1,45 @@
+---
+name: License Check
+
+"on":
+  pull_request:
+    paths:
+      - '**/pyproject.toml'
+      - '.github/workflows/license-check.yml'
+      - '.github/workflows/scripts/check_licenses.py'
+
+jobs:
+  check-licenses:
+    name: Check Package Licenses
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install tomli requests urllib3
+
+      - name: Check licenses
+        run: |
+          python .github/workflows/scripts/check_licenses.py \
+            pyproject.toml || exit_code=$?
+          if [ "${exit_code:-0}" -eq 1 ]; then
+            echo "::error::Found packages with disallowed licenses"
+            exit 1
+          fi
+
+      - name: Check Exchange licenses
+        run: |
+          python .github/workflows/scripts/check_licenses.py \
+            packages/exchange/pyproject.toml || exit_code=$?
+          if [ "${exit_code:-0}" -eq 1 ]; then
```

Comment:
Hi @michaelneale sorry I did not notice we check the exchange licences here.   maybe change here too. 

also one small thing about `${exit_code:-0}`, shall we change to `${exit_code:-1}`? in this case, if the exit_code does not exist, we won't treat it as licence check passing.